scripts/objects/particleSystem.js

```diff
@@ -53,151 +67,103 @@
 
         let api = {
             update: update,
+            get image() { return image; },
             get particles() { return particles; },
+            get isReady() { return isReady; },
+            isDead: isDead
         };
 
         return api;
     }
 
-    function createShipExplosion(xPos, yPos) {
-        console.log('Creating ship explosion at ' + xPos + ': ' + yPos); 
-        effects.push(makeParticleEffect({
-            center: { x: xPos, y: yPos },
-            size: { mean: 12, stdev: 3 },
-            speed: { mean: 65, stdev: 35 },
-            lifetime: { mean: 4, stdev: 1},
-            imageSrc: "resources/textures/fire.png" 
-        }));
-    }
-
-    function createAsteroidBreakup() {
-    console.log('Creating asteroid explosion at ' + xPos + ': ' + yPos); 
-        effects.push(makeParticleEffect({
-            center: { x: xPos, y: yPos },
-            size: { mean: 12, stdev: 3 },
-            speed: { mean: 65, stdev: 35 },
-            lifetime: { mean: 4, stdev: 1},
-            imageSrc: "resources/textures/smoke.png" 
-        }));
-
-    }
-
-    function createUFOExplosion() {
-    console.log('Creating UFO explosion at ' + xPos + ': ' + yPos); 
-        effects.push(makeParticleEffect({
-            center: { x: xPos, y: yPos },
-            size: { mean: 12, stdev: 3 },
-            speed: { mean: 65, stdev: 35 },
-            lifetime: { mean: 4, stdev: 1},
-            imageSrc: "resources/textures/fire.png" 
-        }));
-
-    }
-
-    function update(elapsedTime) {
-        for(let e = 0; e < effects.length; e++) {
-            effects[e].update(elapsedTime); 
-        }
-    }
-    let api = {
-        createShipExplosion: createShipExplosion,
-        createAsteroidBreakup: createAsteroidBreakup,
-        createUFOExplosion: createUFOExplosion,
-        update: update,
-        get effects() { return effects; } 
+    function createHyperspaceEffect(spaceship) {
+        effects.push(makeEffect({
+            center: { x: spaceship.center.x, y: spaceship.center.y },
+            size: { mean: 20, stdev: 4 }, 
+            speed: { mean: 400, stdev: 20 }, 
+            lifetime: { mean: 0.2, stdev: 0.1 }, 
+            explosionLifetime: 0.2, 
+            density: 8, 
+            imageSrc: "resources/images/laser.png"

```

Comment:
This texture could be improved. Also, the textures could be grouped into a folder.